test/unit/bypass_validation_tests.js

```diff
@@ -0,0 +1,239 @@
+          request.reply({
+            results: 't',
+            ok: 1
+          });
+        } catch (e) {
+          close(e);
+        }
+      }
+
+      if (doc.ismaster) {
+        request.reply(Object.assign({}, mock.DEFAULT_ISMASTER));
+      } else if (doc.endSessions) {
+        request.reply({ ok: 1 });
+      }
+    });
+
+    client.connect(function(err, client) {
+      expect(err).to.not.exist;
+      const db = client.db('test');
+      const collection = db.collection('test_c');
+
+      const options = {
+        out: 'test_c',
+        bypassDocumentValidation: bypassDocumentValidation.actual
+      };
+
+      collection.mapReduce(function map() {}, function reduce() {}, options, e => {
+        close(e);
+      });
+    });
+  }
+  // map reduce
+  it('should only set bypass document validation if strictly true in mapReduce', function(done) {
+    testMapReduce({ expected: true, actual: true }, done);
+  });
+
+  it('should not set bypass document validation if not strictly true in mapReduce', function(done) {
+    testMapReduce({ expected: undefined, actual: false }, done);
+  });
+
+  // general test for findAndModify function
+  function testFindAndModify(bypassDocumentValidation, done) {
+    const client = new MongoClient(`mongodb://${test.server.uri()}/test`);
+    let close = e => {
+      close = () => {};
+      client.close(() => done(e));
+    };
+
+    test.server.setMessageHandler(request => {
+      const doc = request.document;
+      if (doc.findAndModify) {
+        try {
+          expect(doc.bypassDocumentValidation).equal(bypassDocumentValidation.expected);
+          request.reply({
+            ok: 1
+          });
+        } catch (e) {
+          close(e);
+        }
+      }
+
+      if (doc.ismaster) {
+        request.reply(Object.assign({}, mock.DEFAULT_ISMASTER));
+      } else if (doc.endSessions) {
+        request.reply({ ok: 1 });
+      }
+    });
+
+    client.connect(function(err, client) {
+      expect(err).to.not.exist;
+      const db = client.db('test');
+      const collection = db.collection('test_c');
+
+      const options = { bypassDocumentValidation: bypassDocumentValidation.actual };
+
+      collection.findAndModify(
+        { name: 'Andy' },
+        { rating: 1 },
+        { $inc: { score: 1 } },
+        options,
+        e => {
+          close(e);
+        }
+      );
+    });
+  }
+  // find and modify
+  it('should only set bypass document validation if strictly true in findAndModify', function(done) {
+    testFindAndModify({ expected: true, actual: true }, done);
+  });
+
+  it('should not set bypass document validation if not strictly true in findAndModify', function(done) {
+    testFindAndModify({ expected: undefined, actual: false }, done);
+  });
+
+  // general test for BlukWrite to test changes made in ordered.js and unordered.js
+  function testBulkWrite(bypassDocumentValidation, done) {
```

Comment:
Nice! One last thing: rename the `bypassDocumentValidation` parameter to `config`.